test/utils/serve/index.test.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-env mocha */
+
+const os = require('os')
+const path = require('path')
+const proxyquire = require('proxyquire')
+const chai = require('chai')
+const createNewProject = require('../../../src/utils/create-new-project')
+
+const temp = os.tmpdir()
+const expect = chai.expect
+const moduleName = '../../../src/utils/serve'
+
+const defaultOptions = {
+  skipGit: true,
+  skipInstall: true,
+  quiet: true
+}
+
+describe('utils/serve', function () {
+  this.timeout(25000)
+
+  it('should run the given server file', (done) => {
+    const appName = `serve-test-${Date.now()}`
+    const appPath = path.join(temp, appName)
+    const serve = proxyquire(moduleName, {
+      '../exec-realtime': (command) => {
+        const expected = path.join(appPath, 'app.js')
+        expect(command).to.equal(`node ${expected}`)
+        done()
+      }
+    })
+
+    createNewProject(temp, 'app', appName, defaultOptions)
```

Comment:
How come promises instead of await?

Reply:
The `serve()` function executes a shell command or starts a server. Both operations don't have an exit and cannot use `await`. I find the callback `done()` is more useful when mocking the dependencies.

But if I use the `done()` function and `async` function at the same time, Mocha complains about it and only allows to either return a promise or use the callback. So I have to use the old-style promise chain but not return the promise.